File: Trabajo_9/CPP/Actividades/Actividad_5.cpp
#include <iostream>
using namespace std;
#include <math.h>

class FiguraGeometrica{
public:
    virtual void calcularArea() = 0;
    virtual void calcularPerimetro() = 0;
    //pi*diametro
};

class Circulo : public FiguraGeometrica{
private:
    int radio;
    int diametro;
public:
    Circulo(int r){
        this -> radio = r;
        this -> diametro = 2*radio;
    }

    void calcularArea() override{
        cout << "Area: " << 3.14 * (radio * radio);
    }

    void calcularPerimetro() override{
        cout << "Perimetro: " << 3.14 * (diametro);
    }
};

class Rectangulo : public FiguraGeometrica{
private:
    int base;
    int altura;
public:
    Rectangulo(int b, int h){
        this -> base = b;
        this -> altura = h;
    }

    void calcularArea() override{
        cout << "Area: " << base*altura;
    }

    void calcularPerimetro() override{
        cout << "Perimetro: " << (base+altura)*2;
    }
};

class Triangulo : public FiguraGeometrica{
private:
    int base;
    int altura;
    int hipotenusa;
public:
    Triangulo(int b, int h){
        this -> base = b;
        this -> altura = h;
        this -> hipotenusa = sqrt((b*b)+(h*h));
    }

    void calcularArea() override{
        cout << "Area: " << (base*altura)/2;
    }

    void calcularPerimetro() override{
        cout << "Perimetro: " << (base+altura+hipotenusa);
    }
};

int main(){
    FiguraGeometrica* circulito = new Circulo(5);
    FiguraGeometrica* rectangulito = new Rectangulo(10,20);
    FiguraGeometrica* triangulito = new Triangulo(7,14);

    FiguraGeometrica *arreglo[3] = {circulito,rectangulito,triangulito};

    for (FiguraGeometrica *p : arreglo){
        cout << "\n" ;
        p->calcularArea();
        cout << "\n" ;
        p->calcularPerimetro();
    }

    return 0;
}

File: Trabajo_9/PY/Actividades/Actividad_5.py
from abc import ABC, abstractmethod
class Figura(ABC):
    def calcularArea(self):
        pass
    def calcularPerimetro(self):
        pass

class Circulo(Figura):
    def __init__(self,radio):
        self.radio=radio
        self.pi=3.1415

    def calcularArea(self):
        print(f"Area de la circulo: {round(self.pi*(self.radio * self.radio),2)}")

    def calcularPerimetro(self):
        print(f"Perimetro de la circulo: {round(2*(self.pi*self.radio),2)}\n")

class Rectangulo(Figura):
    def __init__(self,base,altura):
        self.base=base
        self.altura=altura

    def calcularArea(self):
        print(f"Area de la rectangulo: {self.base*self.altura}")

    def calcularPerimetro(self):
        print(f"Perimetro de la rectangulo: {(self.base+self.altura)*2}\n")

class Triangulo(Figura):
    def __init__(self,base,altura,hipotenusa):
        self.base=base
        self.altura=altura
        self.hipotenusa=hipotenusa

    def calcularArea(self):
        print(f"Area de la triangulo: {self.base*self.altura/2}")

    def calcularPerimetro(self):
        print(f"Perimetro de la triangulo: {self.base+self.altura+self.hipotenusa}\n")

if __name__ == "__main__":
    circulo=Circulo(5)
    triangulo=Triangulo(3,4,5)
    rectangulo=Rectangulo(3,5)
    figuras = [circulo,triangulo,rectangulo]

    for figura in figuras:
        figura.calcularArea()
        figura.calcularPerimetro()

